src/api-v1/paths/preparations/payments.ts

```diff
@@ -0,0 +1,270 @@
+            errors: [
+              {
+                code: ERRORS.CODES.NO_DESTINATION_INSUFFICIENT_XRP,
+                message: 'The account on the receiving end of the payment does not exist',
+                hint: 'Send enough XRP to create the account',
+                url: 'https://xrpl.org/reserves.html',
+                ref: 'tecNO_DST_INSUF_XRP'
+              }
+            ]
+          }
+          throw error;
+        }
+      }
+    }
+
+    const removeUndefined = (obj: any) => {
+      Object.keys(obj).forEach(key => obj[key] === undefined && delete obj[key]);
+      return obj;
+    };
+    const transaction = removeUndefined({
+      TransactionType: "Payment",
+      Account: options.source,
+      Destination: options.destination,
+      Amount
+    });
+
+    try {
+      logger.trace('Preparing', transaction);
+      const prepared = await api.prepareTransaction(transaction);
+      const response = JSON.parse(prepared.txJSON);
+      finishRes(res, 200, response);
+
+    } catch(error) {
+      logger.warn(error);
+      // e.g. ValidationError(instance.Account is not exactly one from <xAddress>,<classicAddress>)
+
+      const status = 400;
+      let message = error.data && error.data.error_message ? error.data.error_message : error.name || 'Error';
+      if (message === 'Account not found.') {
+        message = 'Source account not found.';
+      }
+      if (error.data && error.name) {
+        error.data.name = error.name; // e.g. "RippledError"
+      }
+      error = error.data || error;
+      if (error.code === undefined) {
+        error.code = ERRORS.CODES.GET_PREPARATIONS_PAYMENTS;
+      }
+
+      // Make the error easier to understand
+      error.message = error.message?.replace('instance.', '');
+
+      const response = {
+        message,
+        errors: [error],
+        Transaction: transaction
+      };
+      finishRes(res, status, response); // Validates
+    }
+  }
+
+  const operations = {
+    get
+  };
+
+  return operations as Operations;
+}
+
+export interface ClassicAccountAndTag {
+  classicAccount: string
+  tag: number | false | undefined
+}
+
+/**
+ * Given an address (account), get the classic account and tag.
+ * If an `expectedTag` is provided:
+ * 1. If the `Account` is an X-address, validate that the tags match.
+ * 2. If the `Account` is a classic address, return `expectedTag` as the tag.
+ *
+ * @param Account The address to parse.
+ * @param expectedTag If provided, and the `Account` is an X-address,
+ *                    this method throws an error if `expectedTag`
+ *                    does not match the tag of the X-address.
+ * @returns {ClassicAccountAndTag}
+ *          The classic account and tag.
+ */
+function getClassicAccountAndTag(
+  Account: string,
+  expectedTag?: number
+): ClassicAccountAndTag {
+  if (isValidXAddress(Account)) {
+    const classic = xAddressToClassicAddress(Account)
+    if (expectedTag !== undefined && classic.tag !== expectedTag) {
```

Comment:
&& -> || ?

Reply:
&& is intentional in this case; if there is no `expectedTag`, then there is no need to check that it matches `classic.tag`